src/activities/space-encounter/space-encounter.js

```diff
@@ -0,0 +1,120 @@
+/* GCompris - space-encounter.js
+ *
+ * Copyright (C) 2016 YOUR NAME <xx@yy.org>
+ *
+ * Authors:
+ *   <THE GTK VERSION AUTHOR> (GTK+ version)
+ *   "YOUR NAME" <YOUR EMAIL> (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+.pragma library
+.import QtQuick 2.0 as Quick
+
+var currentLevel = 0
+var numberOfLevel = 1  /**************************************************/
+var items
+var gameWon = false
+
+function start(items_) {
+    items = items_
+    currentLevel = 0
+    initLevel()
+}
+
+function stop() {
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+    //////////////////////////////////////////////////////////////////////////////
+    items.flag = true
+    items.timerRunning = false
+    items.timerRepeat = true
+
+    items.spacePresses = 0
+    items.ticks = 0
+    items.playerScore = 0
+
+    items.textbuttText = "Press Space Bar to Start"
+    items.displayCounterText = items.ticks + " Seconds"
+    items.spaceBarButtonColor = "#ABCDEF"
+    ///////////////////////////////////////////////////////////////////////////////////
+}
+
+function nextLevel() {
+    if(numberOfLevel <= ++currentLevel ) {
+        currentLevel = 0
+    }
+    initLevel();
+}
+
+function previousLevel() {
+    if(--currentLevel < 0) {
+        currentLevel = numberOfLevel - 1
+    }
+    initLevel();
+}
+
+
+
+////////////////////////////////////////////////////////////////////////
+var spaceButtonColor = true;
+
+function increment() {
+    if(items.flag == true) {
+        items.spacePresses = items.spacePresses +1;
+        items.textbuttText = "Your Score " + items.spacePresses;
```

Comment:
you don't need to update it there. It should be binded to the variable.
A better way to do it would be to have in the qml: textButtText = isGameStarted ? qsTr("Your Score %1").arg(items.spacePresses) : qsTr("Press Space Bar to Start")